ci/generate-docs.sh

```diff
@@ -4,8 +4,9 @@ set -e -u -x -o pipefail
 
 cd "$(dirname "$0")/../"
 
+DOCGEN_PIN="${DOCGEN_OVERRIDE:-v1.0}"
 CURRENT_DIR=$(pwd)
-TMP_DIR=$(mktemp -d)
+TMP_DIR=$(mktemp -d "${TMPDIR:-/tmp}/XXXXXXXXX")
```

Comment:
why so many X? what does that do?

Reply:
Controls how many random characters to generate in the path. 8 characters for sufficiently small chances of collision